Assets/Scripts/Effects/GenerateWaterPlane.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+namespace Effects
+{
+    [ExecuteInEditMode]
```

Comment:
```suggestion
    [ExecuteAlways]
```
ExecuteInEditMode will be removed at some point. This replaces it and also fixes problem with scripts that run in EditMode that are placed on Prefabs
https://docs.unity3d.com/ScriptReference/ExecuteInEditMode.html

Reply:
rm [ExecuteInEditMode]